java/src/main/native/src/RmmJni.cpp

```diff
@@ -79,18 +86,46 @@ class tracking_resource_adaptor final : public base_tracking_resource_adaptor {
 
   std::size_t get_total_allocated() override { return total_allocated.load(); }
 
+  std::size_t get_max_total_allocated() override { return max_total_allocated; }
+
+  void reset_scoped_max_total_allocated(std::size_t initial_value) override {
+    scoped_allocated = 0;
+    scoped_max_total_allocated = initial_value;
```

Comment:
I think you need the lock here, to avoid a potential race with line 128.

Reply:
Nice catch!